app/webpacker/controllers/cart_controller.js

```diff
@@ -0,0 +1,20 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ["outOfStockMessage"];
+
+  onQuantityChange(e) {
```

Comment:
It's great that we can provide more direct feedback to the user, but unfortunate that the code duplicates the check that we already do in Ruby (in `Stock::AvailabilityValidator`). 

I'm not yet familiar with Stimulus Reflex, but I assume the intention is that it allows us to make use of the same Ruby code and provide quick feedback to the user. So is there a way that Stimulus Reflex can provide this feedback, without needing to add extra logic in Javascript?

Reply:
I'm not sure I've understand your points, but I'll try to anwser:

 - Here, we're not using [StimulusReflex](https://docs.stimulusreflex.com/) but only [Stimulus](https://stimulus.hotwired.dev/). Stimulus is a very small and simple framework to work with html _we already have_ (unlike React framework for example, which can _generate_ html via `jsx` language). 
 - To me, the only logic we have is this line:
```js
e.currentTarget.value >= e.currentTarget.max
```
`max` is an attribute of the input filled with ruby in a `haml` template. Therefore I don't see any code duplication regarding the business logic. 
Actually this controller (`app/webpacker/controllers/cart_controller.js`) could be renamed into something like `field_number_controller.js`. It's agnostic to `cart`, `order`, ... I haven't done like because I thought it was maybe not possible due to the way we integrate each stimulus controller to DOM element and the structure of the page (one of the drawback of Stimulus). But I can have another look.
 - We can transform this Stimulus controller into a StimulusReflex controller. Each time the user change the quantity, we'll call a Reflex on server side (communication between client and server is then done through websocket). The Reflex could check _in real time_ the _real_ stock (something done via `Stock::AvailabilityValidator`?). This will be for sure an improvement, but I'm not sure this is highly needed.

Reply:
I think you did work out what I meant :)

Thanks for responding, and I agree not necessary at this point, but I think still worth considering for the future.

You're right that the only business logic is that one line. But there are several other lines needed too, and we are now doing the same thing twice: 
1. First on page load, check and show an out-of-stock message (Ruby)
2. Second, if quantity changes, check and show an out-of-stock message (JS)

In principle, StimulusReflex gives us the opportunity to minimise the added JS, which would make the codebase easier to maintain in the long term.